testbook/tests/test_reference.py

```diff
@@ -17,6 +17,20 @@ def test_create_reference(notebook):
     assert a.resolve() == [1, 2, 3]
 
 
+def test_create_reference_getitem(notebook):
+    a = notebook["a"]
+    assert repr(a) == "'[1, 2, 3]'"
+    assert a._type == 'list'
+    assert a.resolve() == [1, 2, 3]
```

Comment:
Should we make `ref` object `==` operator call `resolve` by default?

Reply:
I had initially thought about doing this during the mid-GSoC period last year, but I soon realised that calling resolve by default on the `==` operator will lead to the same "pull up from kernel" issue which we have discussed in depth. 

Also, with the way it is right now, the equality of two non-serializable variables can be asserted using testbook. For example:

```python
# notebook
var1 = var2 = SomeNonSerializableClass() 
```

```python
# test.py
def test_var_equality(tb):
    var1 = tb.ref('var1')
    var2 = tb.ref('var2')
    assert var1 == var2
```

Reply:
Got it